ui/src/test/java/org/pentaho/di/ui/trans/step/BaseStepDialog_ConnectionLine_Test.java

```diff
@@ -74,87 +103,131 @@ public void adds_WhenConnectionNameIsUnique() throws Exception {
 
-    Supplier<Spoon> mockSupplier = mock( Supplier.class );
-    Spoon mockSpoon = mock( Spoon.class );
-    Whitebox.setInternalState( dialog, "spoonSupplier", mockSupplier );
-    when( mockSupplier.get() ).thenReturn( mockSpoon );
-    dialog.transMeta = transMeta;
-    dialog.new AddConnectionListener( mock( CCombo.class ) ).widgetSelected( null );
+    mockDialog.transMeta = transMeta;
+    mockDialog.new AddConnectionListener( mock( CCombo.class ) ).widgetSelected( null );
     if ( answeredName != null ) {
       verify( mockSpoon, times( 1 ) ).refreshTree( anyString() );
     }
   }
 
+  @Test
+  public void edits_globalConnectionWhenNotRenamed() throws Exception {
+    TransMeta transMeta = new TransMeta();
+    DatabaseMeta db = createDefaultDatabase();
+
+    transMeta.addDatabase( db );
+    dbMgr.addDatabase( db );
+    assertTotalDbs( transMeta, 2 );
+
+    invokeEditConnectionListener( transMeta, INITIAL_NAME );
+
+    DatabaseMeta localDb = transMeta.getDatabaseManagementInterface().getDatabase( INITIAL_NAME );
+    DatabaseMeta globalDb =
+      transMeta.getDatabases().stream().filter( databaseMeta -> databaseMeta.getName().equals( INITIAL_NAME ) )
+        .findFirst().get();
+
+    assertEquals( INITIAL_HOST, localDb.getHostname() );
+    assertEquals( INPUT_HOST, globalDb.getHostname() );
+  }
 
   @Test
-  public void edits_WhenNotRenamed() throws Exception {
+  public void edits_localConnectionWhenNotRenamed() throws Exception {
     TransMeta transMeta = new TransMeta();
     transMeta.getDatabaseManagementInterface().addDatabase( createDefaultDatabase() );
 
     invokeEditConnectionListener( transMeta, INITIAL_NAME );
 
-    assertOnlyDbExists( transMeta, INITIAL_NAME, INPUT_HOST );
+    assertOnlyOneActiveDb( transMeta, INITIAL_NAME, INPUT_HOST );
   }
 
   @Test
-  public void edits_WhenNewNameIsUnique() throws Exception {
+  public void edits_GlobalConnectionWhenNewNameIsUnique() throws Exception {
     TransMeta transMeta = new TransMeta();
-    transMeta.getDatabaseManagementInterface().addDatabase( createDefaultDatabase() );
+    dbMgr.addDatabase( createDefaultDatabase() );
 
     invokeEditConnectionListener( transMeta, INPUT_NAME );
 
-    assertOnlyDbExists( transMeta, INPUT_NAME, INPUT_HOST );
+    assertOnlyOneActiveDb( transMeta, INPUT_NAME, INPUT_HOST );
+    assertTotalDbs( transMeta, 1 );
   }
 
   @Test
-  public void ignores_WhenNewNameIsUsed() throws Exception {
+  public void ignores_EditToLocalConnectionWhenNewNameIsNull() throws Exception {
     TransMeta transMeta = new TransMeta();
     transMeta.getDatabaseManagementInterface().addDatabase( createDefaultDatabase() );
 
     invokeEditConnectionListener( transMeta, null );
 
-    assertOnlyDbExists( transMeta, INITIAL_NAME, INITIAL_HOST );
+    assertOnlyOneActiveDb( transMeta, INITIAL_NAME, INITIAL_HOST );
+    assertTotalDbs( transMeta, 1 );
+  }
+
+  @Test
+  public void ignores_EditToGlobalConnectionWhenNewNameIsNull() throws Exception {
```

Comment:
I think we need a new test case that: 
adds a DB to local
edits that DB
validates that the modified DB is still at the local level after the edit.